### ISSUE TITLE
Edge case where cb throws an error

### ISSUE THREAD

**Issue by ArtskydJ** [coding-in-the-wild/just-login-core#3]
https://github.com/coding-in-the-wild/just-login-core/blob/master/core.js#L19

If the callback is called, and then it throws an error, the thrown error will be caught and the the callback will be called with that error.

It could act something like this:
```js
var n = 0
function cb(err, x) {
  n++
  if (x) throw new Error('whoops')
}
try {
  cb(null, true)
} catch (e) {
  cb(e)
}
console.log(n) // probably 2
```

**Comment by TehShrike** [comment]
Good catch!

We shouldn't have a try/catch around the callback invocation, just the JSON.parse:

```js
try {
	var resultObject = JSON.parse(result)
} catch (e) {
	cb(e)
}
cb(null, resultObject)			
```

**Comment by ArtskydJ** [comment]
Fixed by #4.
